hack/release.sh

```diff
@@ -97,11 +97,18 @@ function build_images() {
       ;;
   esac
 
+  # If we are running in Prow, use the Google-hosted goproxy
+  local goproxy="direct"
+  if [ "${PROW_JOB_ID}" ]; then
+    goproxy="https://proxy.golang.org"
```

Comment:
Hmm, one thought on this... Instead of hardcoding this, we could just check if `GOPROXY` is set as an env var (that's the only way to set it), and if it is, pass through it's value. That way it would work the same if it's in Prow or any other environment. It would always respect that setting.

I like the sound of that better... blargh. :)